File: ClassStuff/1_2/1_2.py
import json
from datetime import datetime, timezone
def lambda_handler(event, context):
    requiredFields = ["title", "director_name", "runtime_minutes", "release_year"]
    for field in requiredFields:
        if field not in event:
            return{
                "statusCode" : 400,
                "body" : json.dumbs({"message" : f"Missing required fields : {field}"})
            }
    runtimeMinutes = event["runtime_minutes"]
    if not isinstance(runtimeMinutes, int) or runtimeMinutes <= 0:
        return {
            "statusCode" : 400,
            "body" : json.dumps({"message" : "runtime minutes must be a positive integer"})
        }
    title = event["title"]
    directorName = event["director_name"]
    releaseYear = event["release_year"]
    current_year = datetime.now(timezone.utc).year
    decade = (releaseYear // 10) * 10
    is_classic = releaseYear < 1980
    runtime_hours  = round(runtimeMinutes / 60, 2)
    film_age_years = current_year - releaseYear
    return {
        "statusCode": 200,
        "body": json.dumps({
            "display":         f"{title} ({releaseYear})",
            "director":        directorName,
            "decade":          f"{decade}s",
            "is_classic":      is_classic,
            "runtime_hours":   runtime_hours,
            "film_age_years":  film_age_years,
            "generated_at":    datetime.now(timezone.utc).isoformat()
        })
    }
    #test
def run_local_tests():
    tests = [
        {
            "title" : "Death Stranding 2: On the Beach",
            "director_name" : "Hideo Kojima",
            "runtime_minutes" : 9999,
            "release_year" : 2025
        },{
            "title" : "Death Stranding 2: On the Beach",
            "director_name" : "Hideo Kojima",
            "runtime_minutes" : -1,
            "release_year" : 2025
        },{
            "title" : "The Grinch",
            "director_name" : "Hideo Kojima",
            "runtime_minutes" : 16,
            "release_year" : 90
        },{
            "title" : "Blue",
            "director_name" : "Hideo Kojima",
            "runtime_minutes" : 120,
            "release_year" : 1900
        }
    ]
    for event in tests:
        result = lambda_handler(event, None)
        print(f"Result: {result}")
if __name__ == "__main__":
    run_local_tests()

File: ClassStuff/1_2/1_2.cpp
#include <iostream>
#include <chrono>
#include <string>
#include "include/nlohmann/json.hpp"
namespace ClassWork{
    class OneThree{
        nlohmann::json LambdaHandler(nlohmann::json event){
            std::string requiredFields[] = {"title", "director_name", "runtime_minutes", "release_year"};
            for(std::string field : requiredFields){
                if(!event.contains(field)){
                    nlohmann::json result = {
                        {"statusCode", 400},
                        {"body",
                            {"message", "Missing required fields : " + field}
                        }
                    };
                    return result;
                }
            }
            float runtimeMinutes = event.value("runtime_minutes", 0);
            if(runtimeMinutes <= 0){
                nlohmann::json result = {
                        {"statusCode", 400},
                        {"body",
                            {"message", "runtime minutes must be a positive integer"}
                        }
                    };
                    return result;
            }
            const std::string title = event.value("title", "");
            const std::string directorName = event.value("director_name", "");
            const int releaseyear = event.value("release_year", 0);
            const auto now = std::chrono::system_clock::now();
            const std::chrono::year_month_day ymd{std::chrono::floor<std::chrono::days>(now)};
            const std::time_t t_c = std::chrono::system_clock::to_time_t(now);
            //const currentYear = 
            nlohmann::json result = {
                {"echo", inputText},
                {"reversed", inputText},
                {"length", inputText.size()},
                {"timestamp", std::ctime(&t_c)}
            };
            return result;
        }
        public:
            void RunLocalTests(){
                std::cout<<"Hello World"<<std::endl;
                nlohmann::json test = {
                    {"inputText", "Hello AWS!"},
                    {"inputText", "GoodBye AWS!"},
                    {"inputText", "x"},
                    {}
                };
                for(auto event : test){
                    nlohmann::json result = LambdaHandler(test);
                    std::cout<<"Input : "<<event<<std::endl;
                    std::cout<<"Result : "<<result<<std::endl<<std::endl;
                }
            }
    };
};
int main(){
    ClassWork::OneThree* test = new ClassWork::OneThree();
    test->RunLocalTests();
    delete test;
    return 0;
}